src/libmongoc/tests/TestSuite.c

```diff
@@ -580,6 +589,33 @@ TestSuite_RunTest (TestSuite *suite, /* IN */
       test_msg ("Begin %s, seed %u", name, test->seed);
    }
 
+   for (i = 0; i < suite->failing_flaky_skips.len; i++) {
+      TestSkip *skip =
+         _mongoc_array_index (&suite->failing_flaky_skips, TestSkip *, i);
+      if (0 == strcmp (name, skip->test_name)) {
```

Comment:
```suggestion
      if (0 == strcmp (name, skip->test_name) && skip->subtest_desc == NULL) {
```

I think without this change, this will skip all tests in a JSON test file.

For example, running:
```
./cmake-build/src/libmongoc/test-libmongoc --skip-tests ./.evergreen/skip-tests.txt --no-fork -d -l "/sessions/unified/snapshot-sessions"
```

Currently prints:
```
{ "status": "skip", "test_file": "/sessions/unified/snapshot-sessions", "reason": "error: checking expectResult:  { "$numberInt" : "2" }" }
```

Which indicates all subtests in the file are skipped. The skip-tests.txt specifies to only skip the "countDocuments operation with snapshot" subtest.